identity-wallet/src/subject.rs

```diff
@@ -0,0 +1,101 @@
+    pub secret_manager: SecretManager,
+}
+
+#[async_trait]
+impl Sign for Subject {
+    async fn key_id(&self, subject_syntax_type: &str) -> Option<String> {
+        let method: DidMethod = serde_json::from_str(&format!("{subject_syntax_type:?}")).ok()?;
+
+        self.secret_manager
+            .produce_document(method)
+            .await
+            .ok()
+            .and_then(|document| document.verification_method().first().cloned())
+            .map(|first| first.id().to_string())
+    }
+
+    async fn sign(&self, message: &str, _subject_syntax_type: &str) -> anyhow::Result<Vec<u8>> {
+        Ok(self.secret_manager.sign(message.as_bytes()).await?)
+    }
+
+    fn external_signer(&self) -> Option<Arc<dyn ExternalSign>> {
+        None
+    }
+}
+
+#[async_trait]
+impl oid4vc::oid4vc_core::Subject for Subject {
+    async fn identifier(&self, subject_syntax_type: &str) -> anyhow::Result<String> {
+        let method: DidMethod = serde_json::from_str(&format!("{subject_syntax_type:?}"))?;
+
+        Ok(self
+            .secret_manager
+            .produce_document(method)
+            .await
+            .map(|document| document.id().to_string())?)
+    }
+}
+
+#[async_trait]
+impl Verify for Subject {
+    async fn public_key(&self, did_url: &str) -> anyhow::Result<Vec<u8>> {
+        let did_url = identity_iota::did::DIDUrl::parse(did_url).unwrap();
+
+        let resolver = Resolver::new().await;
+
+        let document = resolver.resolve(did_url.did().as_str()).await.unwrap();
+
+        let verification_method = document
+            .resolve_method(
+                DIDUrlQuery::from(&did_url),
+                Some(identity_iota::verification::MethodScope::VerificationMethod),
+            )
+            .unwrap();
+
+        // Try decode from `MethodData` directly, else use public JWK params.
+        verification_method
+            .data()
+            .try_decode()
+            .or_else(|_| {
+                verification_method
+                    .data()
+                    .public_key_jwk()
+                    .and_then(|public_key_jwk| match public_key_jwk.params() {
+                        JwkParams::Okp(okp_params) => Some(okp_params.x.as_bytes().to_vec()),
+                        JwkParams::Ec(ec_params) => Some(ec_params.x.as_bytes().to_vec()),
+                        _ => None,
+                    })
+                    .ok_or(anyhow::anyhow!("Failed to decode public key for DID URL: {}", did_url))
+            })
+            .and_then(|encoded_public_key| URL_SAFE_NO_PAD.decode(encoded_public_key).map_err(Into::into))
+    }
+}
+
+// Helper function: load a `Subject`
+pub async fn subject(stronghold_manager: Arc<StrongholdManager>, password: String) -> Arc<Subject> {
+    let client_path = crate::persistence::STRONGHOLD
+        .lock()
+        .unwrap()
+        .to_str()
+        .ok_or(anyhow::anyhow!("failed to get stronghold path"))
+        .unwrap()
```

Comment:
It is a bit weird to use anyhow here if you're not intending to return the error. So you could either properly handle + return the error (so you would need to change the return type of this function to `anyhow::Result<Arc<Subject>>`..

..or if you are certain that this can never fail and there is no need to return any error, replace these two lines with `.expect("failed to get stronghold path")`.

Reply:
I changed it to `expect()` since we will always have a Stronghold at this point.